docs: Update the documentation for Card and CardContentControl (#229)
unoplatform/uno.toolkit.ui: doc/controls/CardAndCardContentControl.md

# Card & CardContentControl

> ## Summary

Reviewer: Can we add screenshots?

Author: Yes @jeromelaban but we will add them as the same time that the v2 styles will to be done for Card next week. (cc @kazo0)

## Card
The Card control comes with all the built-in properties of a `Control`  and also a few additional properties listed below that let you tailor the content to be displayed:

| Property                  | Type              |
|---------------------------|-------------------|
| HeaderContent             | object            |
| HeaderContentTemplate     | DataTemplate      |
| SubHeaderContent          | object            |
| SubHeaderContentTemplate  | DataTemplate      |
| AvatarContent             | object            |
| AvatarContentTemplate     | DataTemplate      |
| MediaContent              | object            |
| MediaContentTemplate      | DataTemplate      |
| SupportingContent         | object            |
| SupportingContentTemplate | DataTemplate      |
| IconsContent              | object            |
| IconsContentTemplate      | DataTemplate      |
| Elevation                 | double            |
| ShadowColor               | Windows.UI.Color  |

Consider using [CardContentControl](#cardcontentcontrol) if you need full control over the content layout.

### Usage

```xml
xmlns:utu="using:Uno.Toolkit.UI"
...

> <-- ElevatedCardStyle -->

Reviewer 2: ```suggestion
<!-- ElevatedCardStyle -->
```
also can remove the blank line between comment and the <utu:Card />

Reviewer 2: same for all occurrences

Author: good catch thanks ;)

# Card & CardContentControl

> ## Summary

Reviewer 2: also worth adding an one-liner for the CardContentControl in the summary

Card represents a control identifiable as a single, contained unit used to visually group related child content and actions that relate information about a subject.

> A card’s layout and dimensions depend on its contents.

Reviewer 2: french apostrophe
```suggestion
A card's layout and dimensions depend on its contents.
```

> A card’s layout and dimensions depend on its contents.

Reviewer 2: maybe move this line under ##card?

Author: I decided to change the content to do a summary for both at the same time as they have the same definition at the end

## Remarks

Currently, there are three [Material](https://m3.material.io/components/cards/) styles for `Card` and `CardContentControl` that you can use. 
Depending on the amount of user attention you want to draw to the content you can use:
- `ElevatedCardStyle` or `ElevatedCardContentControlStyle` to add a subtle z-axis elevation.
- `FilledCardStyle` or `FilledCardContentControlStyle` to display a simple background color without any elevation or border for the card.
- `OutlinedCardStyle` or `OutlinedCardContentControlStyle` to display a simple solid stroke along the border of the card.

## Card

> The Card control comes with all the build-in properties deriving from `Control` but also has these additional properties listed below that you can use depending on the content that you want to display:

Reviewer 2: ```suggestion
The Card control comes with all the built-in properties of a `Control`  and also a few additional properties listed below that let you tailor the content to be displayed:
```

| Property                  | Type              |
|---------------------------|-------------------|

> | HeaderContent             | Object            |

Reviewer 2: prefer the lowercase alias
```suggestion
| HeaderContent             | object            |
```
same for `double` below

| HeaderContentTemplate     | DataTemplate      |
| SubHeaderContent          | Object            |
| SubHeaderContentTemplate  | DataTemplate      |
| AvatarContent             | Object            |
| AvatarContentTemplate     | DataTemplate      |
| MediaContent              | Object            |
| MediaContentTemplate      | DataTemplate      |
| SupportingContent         | Object            |
| SupportingContentTemplate | DataTemplate      |
| IconsContent              | Object            |
| IconsContentTemplate      | DataTemplate      |
| Elevation                 | Double            |
| ShadowColor               | Windows.UI.Color  |

> If you want to have something more custom for the entire content you can use  [CardContentControl](#cardcontentcontrol) instead of using `Card`.

Reviewer 2: ```suggestion
Consider using [CardContentControl](#cardcontentcontrol) if you need full control over the content layout.
```

Reviewer 2: "the layout of the content"? not sure which flows better

> ### Usage for Card

Reviewer 2: we are already nesting under ## Card
```suggestion
### Usage
```

Reviewer 2: same for cardcontentcontrol below

> #### ElevatedCardStyle

Reviewer 2: you can merge the 3 example into a single one
and replace the #### headers with \<-- xml comments -->

Reviewer 2: same for cardcontentcontrol below

```xml
xmlns:utu="using:Uno.Toolkit.UI"
...
<utu:Card HeaderContent="Elevated card"
		  SubHeaderContent="With title and subtitle"
		  Style="{StaticResource ElevatedCardStyle}" />
```
#### FilledCardStyle

```xml
xmlns:utu="using:Uno.Toolkit.UI"
...
<utu:Card HeaderContent="Filled card"
		  SubHeaderContent="With title and subtitle"
		  Style="{StaticResource FilledCardStyle}" />
```

#### OutlinedCardStyle

```xml
xmlns:utu="using:Uno.Toolkit.UI"
...
<utu:Card HeaderContent="Outlined card"
		  SubHeaderContent="With title and subtitle"
		  Style="{StaticResource OutlinedCardStyle}" />
```

## CardContentControl

> The CardContentControl is based on ContentControl and allow you to custom the entire content to fit your needs inside the DataTemplate.

Reviewer 2: ```suggestion
The `CardContentControl` is based on `ContentControl` and allows you to customize the entire content through `DataTemplate` to fit your needs.
```
